server/service/campaigns_test.go

```diff
@@ -175,10 +194,20 @@ func TestLiveQueryAuth(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			ctx := viewer.NewContext(context.Background(), viewer.Viewer{User: tt.user})
 
-			_, err := svc.NewDistributedQueryCampaign(ctx, query1ObsCanRun.Query, nil, fleet.HostTargets{})
+			var tms []uint
+			// Testing RunNew is tricky, because RunNew authorization is done, then
```

Comment:
This brings me to the question:
Authorization-wise, why do we need to distinguish between a "new (saved=false) query" and a "(saved=true) query"...

Reply:
Not sure I understand your question, but the way I understand it I'd say it's because we want some users to be allowed to create new queries, while others (e.g. observers) can only run existing ones?

Reply:
That's my understanding too. IIRC, observers can run only queries that are already created and marked as valid for the role (observer can run = true).

Reply:
Ah, gotcha, makes sense; global/team observers cannot run "any" query they want (non-saved).